gnucash/gnome/gnc-split-reg.c

```diff
@@ -1363,7 +1363,8 @@ gsr_default_schedule_handler( GNCSplitReg *gsr, gpointer data )
 
 	if ( theSX )
 	{
-	    gnc_ui_scheduled_xaction_editor_dialog_create(theSX, FALSE);
+	    gnc_ui_scheduled_xaction_editor_dialog_create(gnc_ui_get_main_window (NULL),
```

Comment:
And here you can either query the main_window from the split register (which I believe has its own function to retrieve it) or you can pass the gsr widget as parameter to gnc_ui_get_main_window. NULL should only be used if there's absolutely not widget that can be used to climb up the widget hierarchy to a top-level window.